common/src/main/java/com/graphicsfuzz/common/util/ShaderJobFileOperations.java

```diff
@@ -1186,21 +1201,80 @@ protected static void writeShaderResultToFileHelper(
       fileOps.writeStringToFile(
-          new File(shaderJobResultNoExtension + ".info.json"),
+          shaderJobResultFile,
           infoJson.toString());
 
-      return;
-    }
+      if (referenceShaderResultFile.isPresent()) {
 
-    final File outputImage = new File(shaderJobResultNoExtension + ".png");
+        // We have reference results, so can populate the "comparison_with_reference" property.
+
+        // This maps to a dictionary with up to 4 keys:
+        // - "exact_match", true if and only if the results are identical
+        // - "exactdiff_output", populated only if "exact_match" is false, with the result of
+        //   exact diffing
+        // - "fuzzy_match", present only if "exact_match" is false, and then true if and only if
+        //   the results are similar
+        // - "fuzzydiff_output", present only if "fuzzy_diff" is set, with the result of
+        //   fuzzy diffing.
+
+        final JsonObject computeShaderComparisonWithReference = new JsonObject();
+        try {
+
+          // Check whether the results exactly match those of the reference.
+          final ExecResult exactDiffResult =
+              new ExecHelper().exec(ExecHelper.RedirectType.TO_BUFFER,
+              null, false,
+                  Paths.get(ToolPaths.getPythonDriversDir(),"inspect-compute-results").toString(),
+              "exactdiff",
+              referenceShaderResultFile.get().getAbsolutePath(),
+              shaderJobResultFile.getAbsolutePath());
+          computeShaderComparisonWithReference.addProperty("exact_match",
+              exactDiffResult.res == 0);
+
+          if (exactDiffResult.res != 0) {
+
+            // In the case that we do not have an exact match, store the output obtained by exact
+            // diffing (as it may be useful to inspect).
+            computeShaderComparisonWithReference.addProperty("exactdiff_output",
+                exactDiffResult.stderr.toString());
+
+            // Now perform a fuzzy diff.
+            final ExecResult fuzzyDiffResult =
+                new ExecHelper().exec(
+                    ExecHelper.RedirectType.TO_BUFFER,
+                    null,
+                    false,
+                    Paths.get(
+                        ToolPaths.getPythonDriversDir(),
+                        "inspect-compute-results").toString(),
+                    "fuzzydiff",
+                    referenceShaderResultFile.get().getAbsolutePath(),
+                    shaderJobResultFile.getAbsolutePath());
+            computeShaderComparisonWithReference.addProperty("fuzzy_match",
+                fuzzyDiffResult.res == 0);
+            computeShaderComparisonWithReference.addProperty("fuzzydiff_output",
+                fuzzyDiffResult.stderr.toString());
+          }
+          infoJson.add("comparison_with_reference", computeShaderComparisonWithReference);
+        } catch (InterruptedException interruptedException) {
```

Comment:
I would just add the InterruptedException to the method signature, or at least rethrow it; I don't think it should happen unless the user tries to terminate the server. I also want to ban "printStackTrace" via a Python script or something, but I am lazy.

Reply:
I have added it to the method signature.  I also got rid of another printStackTrace from later in the method.